app/main.py

```diff
@@ -2,24 +2,65 @@
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.name = None
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: object, owner: object) -> int:
+        return instance.name
+
+    def __set__(self, instance: object, value: int) -> None:
+        instance.name = value
```

Comment:
Wrong realization, validation must be into set method
Also instance.name will not return proper value